nixos/modules/services/monitoring/watchdogd.nix

```diff
@@ -0,0 +1,113 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.watchdogd;
+
+  mkPluginOpts = plugin: {
+    enable = mkEnableOption "watchdogd plugin ${plugin}";
+    interval = mkOption {
+      type = types.ints.unsigned;
+      default = 300;
+      description = ''
+        Amount of seconds between every poll.
+      '';
+    };
+    logmark = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to log current stats every poll interval.
+      '';
+    };
+    warning = mkOption {
+      type = types.ints.unsigned;
+      description = ''
+        The high watermark level. Alert sent to log.
+      '';
+    };
+    critical = mkOption {
+      type = types.ints.unsigned;
+      description = ''
+        The critical watermark level. Alert sent to log, followed by reboot or script action.
+      '';
+    };
```

Comment:
While this might mess up the args of this nice `mkOption` function a bit, I think it would be a lot better user experience for this module if we set some defaults for this. As it stands now, the user would have to set all 6 watermark levels manually, unless I'm reading it wrong. Since the function is never exposed to the user, maybe you could just add two more arguments?

These seem to be the upstream defaults: https://github.com/troglobit/watchdogd/blob/6ae714cd22f24e79270296774e2bfb36570d8209/watchdogd.conf#L92-L165

Reply:
I skipped this because of the loadavg stuff - it's based on the core count as far as I can tell.
But I added defaults for the two others with the extra two arguments.